clean-modular-code/activity-3/clean-code-activity-3.md

```diff
@@ -78,21 +78,69 @@ slideshow:
 # This is the preferred way to catch an error 
 def clean_title(title):
     """
-    It's more Pythonic to try first and then ask for forgiveness later. 
-    If you are writing tests this also makes your code easier to test. 
+    Attempts to return the first character of the title.
+    Raises the same error with a friendly message if the input is invalid.
+    """
+    try:
+        return title[0]
+    except IndexError as e:
+        raise IndexError(f"Oops! You provided a title in an unexpected format. "
+                         f"I expected the title to be provided in a list and you provided "
+                         f"a {type(title)}.") from e
+
+# Example usage:
+title = ""
+print(clean_title(title))  # This will raise an IndexError with the friendly message
+```
+
+```{code-cell} ipython3
+# This is the preferred way to catch an error 
+def clean_title(title):
+    """
+    Attempts to return the first character of the title.
+    Raises the same error with a friendly message if the input is invalid.
     """
     try:
         return title[0]
-    except (TypeError, IndexError):
-        return title
+    except IndexError as e:
+        raise IndexError(f"Oops! You provided a title in an unexpected format. I expected the title to be provided in a list and you provided a {type(title)}.") 
+
+# Example usage:
+title = ""
+print(clean_title(title))  # This will raise an IndexError with the friendly message
+```
+
+If you wish, you can shorten the amount of information returned in the exception by adding `from None` to your exception. This will look nicer to a user but you lose some information in the exception feedback. 
```

Comment:
```suggestion
If you wish, you can shorten the amount of information returned in the error by adding `from None` when you raise the error. This will look nicer to a user, but you lose some detail in the error traceback. 
```